pulsator4droid/src/main/java/pl/bclogic/pulsator4droid/library/circle/CirclePulsatorLayout.java

```diff
@@ -0,0 +1,392 @@
+                animator.cancel();
+            }
+            mAnimators = null;
+        }
+    }
+
+    @Override
+    public void start() {
+
+        if (mAnimators == null) {
+            reset();
+            if (mAnimators == null) {
+                return;
+            }
+        }
+
+        if (mIsStarted) {
+            return;
+        }
+
+        for (int x = 0; x < mAnimators.size(); x++) {
+            ObjectAnimator objectAnimator = (ObjectAnimator) mAnimators.get(x);
+
+            if (!mStartFromScratch) {
+                // instead of delaying the animation, fast-forward it
+                long delay = objectAnimator.getStartDelay();
+                objectAnimator.setStartDelay(0);
+
+                // This is where it starts to get tricky. The documentation of
+                // ValueAnimator#setCurrentPlayTime(long) is a bit confusing about whether it
+                // should be called before or after starting the animation itself. The truth is,
+                // it seems the behavior differs between Android versions. If it gets called at
+                // a wrong time, only some of the animations will start while others will not or
+                // the animated object won't be visible at all.
+                boolean shouldStartBeforeSettingCurrentTime =
+                        Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP_MR1;
+
+                if (shouldStartBeforeSettingCurrentTime) {
+                    objectAnimator.start();
+                }
+                objectAnimator.setCurrentPlayTime(mDuration - delay);
+                if (!shouldStartBeforeSettingCurrentTime) {
+                    objectAnimator.start();
+                }
+            } else {
+                objectAnimator.start();
+            }
+        }
+    }
+
+    @Override
+    public void stop() {
+
+        if (mAnimators == null || !mIsStarted) {
+            return;
+        }
+        for (Animator animator : mAnimators) {
+            animator.end();
+        }
+    }
+
+    @Override
+    protected void reset() {
+        boolean isStarted = isStarted();
+
+        stop();
+        clear();
+        build();
+
+        if (isStarted) {
+            start();
+        }
+    }
+
+    /**
+     * Get current interpolator type used for animating.
+     *
+     * @return Interpolator type as int
+     */
+    public int getInterpolator() {
+        return mInterpolator;
+    }
+
+    /**
+     * Set current interpolator used for animating.
+     *
+     * @param type Interpolator type as int
+     */
+    public void setInterpolator(int type) {
+        if (type != mInterpolator) {
+            mInterpolator = type;
+            reset();
+            invalidate();
+        }
+    }
+
+    /**
+     * Set how many times the pulse should repeat.
+     *
+     * @param repeat {@link #INFINITE} for infinite repeat. Also the default value.
+     */
+    public void setRepeat(int repeat) {
```

Comment:
setPulseCount or setPulseRepeatCount would better describe the setter

Reply:
These are the original library methods just moved to circular pulse implementation, but I can update the name.

Reply:
ok, then it is better to keep them the same